### PR TITLE
feat: added ConfigService service class to evaluate configurations.

### LINKED ISSUE #1041: [ERC Registry] Add mechanism to evaluate STARTING_POINT variable
### Problem  
The `STARTING_POINT` environment variable, as outlined in the design document, determines the starting point for the indexing process, allowing the indexer to avoid re-indexing the entire network. The `STARTING_POINT` can take one of the following forms:  
- Contract ID  
- Contract Address  
- `next` pointer  

### Solution  
Implement a utility function to evaluate the `STARTING_POINT` variable, ensuring the indexer operates as intended and aligns with the defined starting point.

### PR DESCRIPTION
**Pull request by quiet-node**
**Description**:
This PR introduces a `ConfigService` class that provides the following functionality:  

- **`validateConfigs()`**: Ensures the configurations are properly set, specifically validating `HEDERA_NETWORK` and `STARTING_POINT`.  
- **`resolveStartingPoint()`**: Determines and resolves the value of `STARTING_POINT`, which only accepts formats like `0.0.{contract_id}`, `0x{evm_address}`, or `next` pointer. Regardless of the input format, `resolveStartingPoint()` converts and returns the appropriate `next` pointer format, compatible with the `/contracts` mirror node endpoint.  

**Related issue(s)**:

Fixes #1041

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: tools/erc-repository-indexer/erc-contract-indexer/src/services/contractScanner.ts [hunk 1]
```diff
@@ -35,9 +35,8 @@ export class ContractScannerService {
    */
   private readonly mirrorNodeBaseUrl: string;
 
-  constructor() {
-    const network = process.env.HEDERA_NETWORK || 'previewnet';
-    this.mirrorNodeBaseUrl = `https:${network}.mirrornode.hedera.com`;
+  constructor(hederaNetwork: string) {
+    this.mirrorNodeBaseUrl = `https:${hederaNetwork}.mirrornode.hedera.com`;
```
Comment (Nana-EC) [inline tools/erc-repository-indexer/erc-contract-indexer/src/services/contractScanner.ts:39]: I would make mirror node url a config.
Mainnet for example is mainnet-public and not mainnet.
Plus if you do that you allow for custom mirror node urls

File: tools/erc-repository-indexer/erc-contract-indexer/src/utils/constants.ts [hunk 2]
```diff
@@ -23,4 +23,11 @@ export default {
   GET_CONTRACT_ENDPOINT: '/api/v1/contracts',
   ERC_20_JSON_FILE_NAME: 'erc-20.json',
   ERC_721_JSON_FILE_NAME: 'erc-721.json',
+  NETWORK_REGEX: /^(localnet|previewnet|testnet|mainnet)$/,
+  STARTING_POINT_REGEX:
+    /^(0x[a-fA-F0-9]{40}|0\.0\.\d+|\/api\/v1\/contracts\?limit=100&order=asc&contract\.id=gt:0\.0\.\d+)$/,
+  EVM_ADDRESS_REGEX: /^0x[a-fA-F0-9]{40}$/,
+  HEDERA_CONTRACT_ID_REGEX: /^0\.0\.\d+$/,
+  NEXT_POINTER_REGEX:
```
Comment (Nana-EC) [inline tools/erc-repository-indexer/erc-contract-indexer/src/utils/constants.ts:31]: This is more of a getContractsListNext regex.
Just next seems vague and wasn't clear till I saw the actual regex details

File: tools/erc-repository-indexer/erc-contract-indexer/src/services/config.ts [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+      );
+    }
+  }
+
+  /**
+   * Resolves the starting point for indexing based on the configuration.
+   * @returns {Promise<string | null>} A promise that resolves to the starting point string or null if not set.
+   */
+  async resolveStartingPoint(): Promise<string | null> {
+    if (constants.NEXT_POINTER_REGEX.test(this.startingPoint)) {
```
Comment (Nana-EC) [inline tools/erc-repository-indexer/erc-contract-indexer/src/services/config.ts:75]: I think there's a missing test for this flow.
Can you confirm and add one if it's indeed missing

**Review by Nana-EC — COMMENTED** [review]
Looking good.
Some comments

File: tools/erc-repository-indexer/erc-contract-indexer/src/services/contractScanner.ts (hunk 1 above)
Comment (quiet-node) [reply tools/erc-repository-indexer/erc-contract-indexer/src/services/contractScanner.ts:39]: Gotcha I agree and will restore the MIRROR_NODE_URL config.

> Mainnet for example is mainnet-public and not mainnet.

Hmm I believe the correct address is still mainnet.mirrornode.hedera.com as they are documented on the  doc page at https://docs.hedera.com/hedera/sdks-and-apis/rest-api and the API documentation at https://mainnet.mirrornode.hedera.com/api/v1/docs/.

Comment (quiet-node) [reply tools/erc-repository-indexer/erc-contract-indexer/src/services/contractScanner.ts]: Restored MIRROR_NODE_URL config.

File: tools/erc-repository-indexer/erc-contract-indexer/src/services/config.ts (hunk 3 above)
Comment (quiet-node) [reply tools/erc-repository-indexer/erc-contract-indexer/src/services/config.ts]: Ha absolutely great catch! I indeed missed the test for this one flow. Hmm how strange!

Anywho just updated and add a new UT for the flow. Thanks for pointing it out!

File: tools/erc-repository-indexer/erc-contract-indexer/src/utils/constants.ts [hunk 4]
```diff
@@ -23,4 +23,13 @@ export default {
   GET_CONTRACT_ENDPOINT: '/api/v1/contracts',
   ERC_20_JSON_FILE_NAME: 'erc-20.json',
   ERC_721_JSON_FILE_NAME: 'erc-721.json',
+  NETWORK_REGEX: /^(localnet|previewnet|testnet|mainnet)$/,
+  MIRROR_NODE_URL_REGEX:
+    /^https:\/\/(previewnet|testnet|mainnet)\.mirrornode\.hedera\.com$/,
```
Comment (Nana-EC) [inline tools/erc-repository-indexer/erc-contract-indexer/src/utils/constants.ts:28]: Note: mainnet mirror node url is `https://mainnet-public.mirrornode.hedera.com/api/v1/docs/`

File: tools/erc-repository-indexer/erc-contract-indexer/src/utils/helper.ts [hunk 5]
```diff
@@ -54,4 +54,15 @@ export class Helper {
...
+
+  /**
+   * Builds a starting point URL for fetching contracts from the mirror node API.
+   * The URL is constructed to retrieve contracts with IDs greater than the specified contract ID.
+   *
+   * @param {string} contractId - The contract ID to use as a reference for the starting point.
+   * @returns {string} The constructed starting point URL for the mirror node API.
+   */
+  static buildStartingPoint(contractId: string): string {
+    return `/api/v1/contracts?limit=100&order=asc&contract.id=gt:${contractId}`;
```
Comment (Nana-EC) [inline tools/erc-repository-indexer/erc-contract-indexer/src/utils/helper.ts:66]: nit: make string a `const`

**Review by Nana-EC — APPROVED** [review]
Looks good, minor non-blocking suggestions

File: tools/erc-repository-indexer/erc-contract-indexer/src/utils/constants.ts (hunk 4 above)
Comment (quiet-node) [reply tools/erc-repository-indexer/erc-contract-indexer/src/utils/constants.ts:28]: Okay will update this to mainnet-public in the follow-up PR

File: tools/erc-repository-indexer/erc-contract-indexer/src/utils/helper.ts (hunk 5 above)
Comment (quiet-node) [reply tools/erc-repository-indexer/erc-contract-indexer/src/utils/helper.ts:66]: Okay will update this to mainnet-public in the follow-up PR

**Pull request merged**